src/components/Edge.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import CSS from 'csstype';
+
+interface Position {
+  x: number;
+  y: number;
+}
+
+function calculateEdgePositionUsingPythagoras(from: Position, to: Position) {
+  const [first, second] = from.x < to.x ? [from, to] : [to, from];
+  const a = second.y - first.y;
+  const b = second.x - first.x;
+  // c^2 = a^2 + b^2
+  const c = Math.sqrt((a * a) + (b * b));
+  // beta = arccos( b^2 + c^2 - a^2 / 2bc ) and radians to degrees
+  const beta = Math.acos((b * b + c * c - a * a) / (2 * b * c)) * (180/Math.PI);
+  const isSecondHigher = (first: Position, second: Position) =>
```

Comment:
Imho this is confusing, a regular if statement would do the job.

Often less lines != cleaner code